packages/@storefront/sayt/src/previous-search-terms/index.ts

```diff
@@ -0,0 +1,37 @@
+import { provide, tag, Events, Selectors, Store, Tag } from '@storefront/core';
+import Sayt from '../sayt';
+
+@provide('previousSearch', (props) => props)
+@tag('gb-sayt-previous-search', require('./index.html'))
+class PreviousSearch {
+  props: PreviousSearch.Props = {
+    onClick: (query) => () => this.actions.search(query),
```

Comment:
I think you have an extra function here. In this case, invoking `onClick` will return a function. In order to dispatch the `search` action, this second function would *also* need to be invoked.

Reply:
I believe this should read:

```
onClick: (query) => this.actions.search(query),
```

Reply:
@jrmykolyn it's using currying because we don't want the click handler to use the event we want it to use the list item value